drivers/tty/serial/8250/8250_dma.c

```diff
@@ -217,8 +229,9 @@ int serial8250_request_dma(struct uart_8250_port *p)
 	}
 
 	/* TX buffer */
+	dma->tx_buf = kmalloc(UART_XMIT_SIZE, GFP_KERNEL);
```

Comment:
NULL check is missed.

Reply:
Am I right that in that case I also need to tear down the preceding
````
dma->rx_buf = dma_alloc_coherent(dma->rxchan->device->dev, dma->rx_size,
	&dma->rx_addr, GFP_KERNEL);
```
?

Reply:
Assuming yes, so added an additional goto.